### ISSUE TITLE
Can chain 2 but not 3 RGB panels (P5 64x32 1/16)

### ISSUE THREAD

**Issue by davecorbett1961** [2dom/PxMatrix#283]
Successfully chaining 2 panels but not 3.  Posting the code for 3 panels below - code for 2 panels is virtually the same except for these lines:
PxMATRIX display(192,32,P_LAT, P_OE,P_A,P_B,P_C,P_D);
display.setPanelsWidth(3);

## Rules
1. Post your code 
#include <PxMatrix.h>
#include <Adafruit_GFX.h>
#include <ESP8266WiFi.h>        // Include the Wi-Fi library
#include <WiFiUdp.h>

#include <Fonts/FreeSans18pt7b.h>
#include <Fonts/FreeSans9pt7b.h>
#include <Fonts/FreeSans12pt7b.h>
#include <Fonts/FreeSans24pt7b.h>

#define PORT_NUM 8888
#define BUFFERMAX 20

const char *ssid = "scoreboard"; // The name of the Wi-Fi network that will be created
const char *password = "cricket1";   // The password required to connect to it, leave blank for an open network

 WiFiUDP Udp;

 byte packetBuffer[BUFFERMAX]; //buffer to hold incoming packet

// Pins for LED MATRIX
#ifdef ESP32

#define P_LAT 22
#define P_A 19
#define P_B 23
#define P_C 18
#define P_D 5
#define P_E 15
#define P_OE 2
hw_timer_t * timer = NULL;
portMUX_TYPE timerMux = portMUX_INITIALIZER_UNLOCKED;

#endif

#ifdef ESP8266

#include <Ticker.h>
Ticker display_ticker;
#define P_LAT 16
#define P_A 5
#define P_B 4
#define P_C 15
#define P_D 12
#define P_E 0
#define P_OE 2

#endif

#define matrix_width 64
#define matrix_height 32

// This defines the 'on' time of the display is us. The larger this number,
// the brighter the display. If too large the ESP will crash
uint8_t display_draw_time=60; //30-70 is usually fine

//PxMATRIX display(32,16,P_LAT, P_OE,P_A,P_B,P_C);
PxMATRIX display(192,32,P_LAT, P_OE,P_A,P_B,P_C,P_D);
//PxMATRIX display(64,64,P_LAT, P_OE,P_A,P_B,P_C,P_D,P_E);

// Some standard colors
uint16_t myRED = display.color565(255, 0, 0);
uint16_t myGREEN = display.color565(0, 255, 0);
uint16_t myBLUE = display.color565(0, 0, 255);
uint16_t myWHITE = display.color565(255, 255, 255);
uint16_t myYELLOW = display.color565(255, 255, 0);
uint16_t myCYAN = display.color565(0, 255, 255);
uint16_t myMAGENTA = display.color565(255, 0, 255);
uint16_t myBLACK = display.color565(0, 0, 0);

uint16_t myCOLORS[8]={myRED,myGREEN,myBLUE,myWHITE,myYELLOW,myCYAN,myMAGENTA,myBLACK};

#ifdef ESP8266
// ISR for display refresh
void display_updater()
{
  display.display(display_draw_time);
}
#endif

#ifdef ESP32
void IRAM_ATTR display_updater(){
  // Increment the counter and set the time of ISR
  portENTER_CRITICAL_ISR(&timerMux);
  display.display(display_draw_time);
  portEXIT_CRITICAL_ISR(&timerMux);
}
#endif

void display_update_enable(bool is_enable)
{

#ifdef ESP8266
  if (is_enable)
    display_ticker.attach(0.004, display_updater);
  else
    display_ticker.detach();
#endif

#ifdef ESP32
  if (is_enable)
  {
    timer = timerBegin(0, 80, true);
    timerAttachInterrupt(timer, &display_updater, true);
    timerAlarmWrite(timer, 4000, true);
    timerAlarmEnable(timer);
  }
  else
  {
    timerDetachInterrupt(timer);
    timerAlarmDisable(timer);
  }
#endif
}

void setup() {

  Serial.begin(115200);
  delay(10);
  Serial.println('\n');
 
  // Define your display layout here, e.g. 1/8 step, and optional SPI pins begin(row_pattern, CLK, MOSI, MISO, SS)
  display.begin(16);
  display.setPanelsWidth(3);
  display.clearDisplay();
  display.setFont();
  display.setTextWrap(false); //  don't wrap at end of line
  display.setTextColor(myCYAN);
  display.setCursor(0,0);
  display.print("Scoreboard");
  display_update_enable(true);
  display.setTextColor(myMAGENTA);
  display.setCursor(64,0);
  display.print("Scoreboard");
  display.setTextColor(myWHITE);
  display.setCursor(128,0);
  display.print("Scoreboard");

  WiFi.softAP(ssid, password);             // Start the access point
  Serial.print("\nAccess Point \"");
  Serial.print(ssid);
  Serial.println("\" started");

  Serial.print("IP address:\t");
  Serial.println(WiFi.softAPIP());         // Send the IP address of the ESP8266 to the computer
  Udp.begin(PORT_NUM);

  display.setTextColor(myYELLOW);
  display.setCursor(0,16);
  display.print(WiFi.softAPIP());
  display.setCursor(0,0);

}

void loop() {

  get_udp_data();

}

void get_udp_data()
{

  // If there's data available, read a packet
  // Expecting: x, y, height, colour, chars (1-16)
  byte packetSize = Udp.parsePacket();
  if (packetSize) 
  {
    Serial.print("\nUDP data (");
    Serial.print(packetSize);
    Serial.print(") received: ");
    // read the packet into packetBufffer
    Udp.read(packetBuffer, BUFFERMAX);
    
    for (int j=0; j<packetSize; j++)
    {
      Serial.print(packetBuffer[j]);
      Serial.print(' ');
    }
    if (packetBuffer[4]==0)
      test_leds();
    else
    {
      display_packetBuffer(packetSize);
    }

  }
}

void display_packetBuffer (byte len)
{
  byte x = packetBuffer[0];
  byte y = packetBuffer[1];
  byte height = packetBuffer[2];
  byte colour = packetBuffer[3];
  byte char_count = len - 4; 
  String s=""; 
  for (byte i=4; i<len; i++)
  {
    s=s+String(char(packetBuffer[i]));
  }
  display.setFont(&FreeSans18pt7b);
  display.clearDisplay();
  display.setCursor(0,25);
  display.print(s);
}

void test_leds()
{
  String s; char ch;
  display.setFont(&FreeSans18pt7b);
  for (char c='A'; c<'['; c++)
  {
    display.clearDisplay();
    display.setCursor(0,25);
    ch = c+32;
    s = String(c)+String(ch);
    display.print(s);
    delay(900);
  }
  display.clearDisplay();
  display.setCursor(0,25);
  display.print("OK");   
}

void get_charset_info()
{
  int16_t  x1, y1;
  uint16_t w, h;
  for (char c='A'; c<'['; c++)
  {
    display.getTextBounds(String(c), 1, 31, &x1, &y1, &w, &h);
    Serial.print("\n"); Serial.print(c);Serial.print(" width= "); Serial.print(w); Serial.print(" height= "); Serial.print(h);
  }
  for (char c='0'; c<':'; c++)
  {
    display.getTextBounds(String(c), 1, 31, &x1, &y1, &w, &h);
    Serial.print("\n"); Serial.print(c);Serial.print(" width= "); Serial.print(w); Serial.print(" height= "); Serial.print(h);
  }
      for (char c='a'; c<'{'; c++)
  {
    display.getTextBounds(String(c), 1, 31, &x1, &y1, &w, &h);
    Serial.print("\n"); Serial.print(c);Serial.print(" width= "); Serial.print(w); Serial.print(" height= "); Serial.print(h);
  }
}
2. Post a picture of the problem and describe what you expect to see
See photos below of 2 panels working fine but 3 definitely not!
![image](https://user-images.githubusercontent.com/83502899/130233868-713ae0b3-7067-4a6e-9193-f86bd27b0e4b.png)
![image](https://user-images.githubusercontent.com/83502899/130233895-0ade6b31-6f31-4aa1-b968-981c233ace6e.png)

3. Run the pattern test and post the result as gif/video (need to have a bit more than one full RED/YELLOW/WHITE cycle)
Seems to run perfectly!
![image](https://user-images.githubusercontent.com/83502899/130237006-d52193c3-2822-41a2-9452-694bfdbb2282.png)

4. State what version of PxMatrix you are running and what MicroController you use
  PxMatrix 1.8.1 on ESP8266 ESP-12 WeMos D1 Mini compatible

**Comment by 2dom** [comment]
Yeah ... This is a known limitation. Looping the clock and pixel data through so many registers eventually leads to a misalignment of clock and data. I suggest to use ESP32 and a library that uses a parallel connection to the display (best case with DMA)
